dev-app/src/api/api.ts

```diff
@@ -141,17 +150,27 @@ export class Api {
       }
     }
 
-    // TODO: implement connect functionality to set these values
-    // if (
-    //   this.connectedAccount &&
-    //   this.connectedAccount.type === ConnectChargeType.Destination
-    // ) {
-    //   formData.append('on_behalf_of', this.connectedAccount.id);
-    //   formData.append('transfer_data[destination]', this.connectedAccount.id);
-    // }
+    if (
+      this.currentAccount &&
+      this.currentAccount.connectedAccountType === ChargeType.DestinationCharges
+    ) {
+      formData.append('on_behalf_of', this.currentAccount.stripeAccountID);
+      formData.append(
+        'transfer_data[destination]',
+        this.currentAccount.stripeAccountID
+      );
```

Comment:
For destination charges we already have the Connect Account option on the CollectCardPaymentScreen. The values for `on_behalf_of` and `transfer_data[destination]` should come from the inputValues there. 

<img width="403" alt="Screenshot 2023-07-24 at 6 40 54 PM" src="https://github.com/stripe/stripe-terminal-react-native/assets/69359432/73d5e2b2-ee2d-4686-aef1-d077cf25ac21">

we can pass them here with inputValues.connectedAccountId https://github.com/stripe/stripe-terminal-react-native/blob/6f7d6353/dev-app/src/screens/CollectCardPaymentScreen.tsx#L135-L141